### ISSUE TITLE
camelCase Resolver not always Default?

### ISSUE THREAD

**Issue by karelz** [aspnet/AspNetCore#4268]
_From @ArgTang on November 12, 2018 11:13_

# Description

I am trying to create an api call to Slack with a typed `httpClient`. Slack gives a statuscode of `400 ` when formatting with PascalCase.
Although in the documentation it says that .net core is using **Camelcase** as default, my payload is serialized with **PascalCase**, and i manually have to serialize correctly. This feels redundant, I have tried setting `contractresolver` in startup, but it does not help for this.

 ```C#
services.AddMvc(options => options.Filters.Add(new AutoValidateAntiforgeryTokenAttribute()))
                .AddJsonOptions(options => {
                    var settings = options.SerializerSettings;
                    settings.NullValueHandling = NullValueHandling.Ignore;
                    settings.DefaultValueHandling = DefaultValueHandling.Ignore;
                    settings.ContractResolver = new CamelCasePropertyNamesContractResolver();
                })
                .SetCompatibilityVersion(CompatibilityVersion.Version_2_1);
```

The documentation for `PostAsJsonAsync` is as follows:
```C#
        // Remarks:
        //     This method uses a default instance of System.Net.Http.Formatting.JsonMediaTypeFormatter.
        public static Task<HttpResponseMessage> PostAsJsonAsync<T>(this HttpClient client, string requestUri, T value);
```

At first glance it seems good, pick the current apps set `Contractresolver`. But when i breakpoint it seems like it is using `JsonContractResolver` with no `Namingstrategy` set. 
Which makes it understandable why it does **not** serialize with camelcase, but lessens the usefulness of the extentionmethod provided. If this is the case i would ask if it was possible to make an api wich sets the current `SerializerSetting` and posts an object with that.

Is there another method i can use for this?

## system info:
Local runtime 2.1.500-preview-009404
Target Framework: 2.1

## sample
```C#
    public class SlackClient
    {
        private readonly HttpClient _client;

        public SlackClient(HttpClient client, string baseUrl)
        {
            _client = client;
            _client.BaseAddress = baseUrl;
        }

        public async Task<(HttpStatusCode status, string message)> DeliverMessageAsyncWork(SlackMessage payload)
        {
//**************
// I have to manually do it like this, to serialize the object correctly a.k.a as camelCase
//***************
            var settings = new JsonSerializerSettings
            {
                ContractResolver = new CamelCasePropertyNamesContractResolver()
            };
            var camelCaseLoad = JsonConvert.SerializeObject(payload, settings);
            var content = new StringContent(camelCaseLoad, Encoding.UTF8, "application/json");

            var res = await _client.PostAsync("", content);

            if (!res.IsSuccessStatusCode)
            {
                return (res.StatusCode, $"Status: {res.ReasonPhrase}");
            }
            return (status: HttpStatusCode.OK, message: $"Status: {HttpStatusCode.OK}");
        }

        public async Task<(HttpStatusCode status, string message)> DeliverMessageAsyncFail(SlackMessage payload)
        {
// ***********
// This will post object as PascalCase (standard C# obj)
// **************
            var res = await _client.PostAsJsonAsync("", payload);

            if (!res.IsSuccessStatusCode)
            {
                return (res.StatusCode, $"Status: {res.ReasonPhrase}");
            }
            return (status: HttpStatusCode.OK, message: $"Status: {HttpStatusCode.OK}");
        }
    }
```

_Copied from original issue: dotnet/corefx#33408_

**Comment by mkArtakMSFT** [comment]
Thanks for contacting us, @ArgTang.
@dougbu, can you please look into this? Thanks!

**Comment by dougbu** [comment]
@ArgTang we moved this issue to aspnet/AspNetCore to give this question a bit more visibility. The [aspnet/AspNetWebStack](https://github.com/aspnet/AspNetWebStack) repo covers one of the packages you're using while the others are built from sources in this repo (or very soon to be here -- moving from [aspnet/Mvc](https://github.com/aspnet/Mvc)).

You're mixing components that are not directly aware of each other:
- Microsoft.AspNetCore.Mvc.Formatters.Json.nupkg from [aspnet/Mvc](https://github.com/aspnet/Mvc) / this repo includes code that defines `MvcJsonOptions` and the `AddJsonOptions(...)` extension method.
- The System.Net.Http.Formatting.dll assemblies in the Microsoft.AspNet.WebApi.Formatting.nupkg package from the [aspnet/AspNetWebStack](https://github.com/aspnet/AspNetWebStack) repo include .NET Core-compatible code that should work fine in ASP.NET Core applications such as yours. However nothing from that repo depends on ASP.NET Core packages.

Bottom line, you've already done the right thing to configure JSON serialization options for System.Net.Http.Formatting.dll separately.
